src/app/core/services/logging.service.ts

```diff
@@ -7,28 +7,117 @@ import { ElectronLog } from 'electron-log';
 })
 export class LoggingService {
   private logger: ElectronLog | Console;
+  private isInSession = false;
+  private readonly LOG_KEY = 'CATcher-Log';
+  private readonly LOG_FILE_NAME = 'CATcher-log.txt';
+  private readonly LOG_START_HEADER = '====== New CATcher Session Log ======';
+  private readonly LOG_COUNT_LIMIT = 4;
 
   constructor(electronService: ElectronService) {
     if (electronService.isElectron()) {
       this.logger = window.require('electron-log');
     } else {
       this.logger = console;
     }
+
+    this.startSession();
+  }
+
+  reset() {
+    this.isInSession = false;
+  }
+
+  /**
+   * Configures loggging Session if Logging Service is not
+   * in session.
+   */
+  startSession() {
+    if (!this.isInSession) {
+      this.isInSession = true;
+      this.initializeLogCache();
+    }
+  }
+
+  initializeLogCache() {
+    this.setCachedLog(this.getTrimmedLogCache(this.getCachedLog(), this.LOG_COUNT_LIMIT));
+  }
+
+  /**
+   * Trims the existing Log in the browser's cache to a select number
+   * of Sessions if necessary.
+   * @param sessionCount The number of Session Logs to preserve in the cache
+   */
+  getTrimmedLogCache(currentLog: string, sessionCount: number): string {
+    const sessionLogSeparator: string = '\n'.repeat(2); // More new-lines added for clarity.
+
+    // Check if Trimming is Necessary
+    const numberOfSessions: number = currentLog == null ? 0 : currentLog.split('\n')
+      .filter((currentLogLine: string) => currentLogLine.includes(this.LOG_START_HEADER))
+      .length;
+
+    if (numberOfSessions < sessionCount) {
+      return `${numberOfSessions ? `${currentLog}${sessionLogSeparator}` : ''}${this.LOG_START_HEADER}`;
```

Comment:
This statement is a little hard to understand, especially since `numberOfSessions` (a number) is being used in a conditional.

I think it'll be clearer to replace this with an `if-else` statement.
The case where `numberOfSessions` = 0 can be handled on its own, without any comparison to `sessionCount`.

Reply:
Yes sure 👍